test/unit/test_files.py

```diff
@@ -125,9 +126,24 @@ def test_download_and_and_extract_source_dir(move, rmtree, s3_download):
 @patch('sagemaker_containers._files.s3_download')
 @patch('os.path.isdir', lambda x: False)
 @patch('shutil.copy2')
-def test_download_and_and_extract_file(copy, s3_download):
-    uri = _env.channel_path('code')
+def test_download_and_extract_file(copy, s3_download):
+    uri = __file__
     _files.download_and_extract(uri, _env.code_dir)
 
     s3_download.assert_not_called()
     copy.assert_called_with(uri, _env.code_dir)
+
+
+@patch('sagemaker_containers._files.s3_download')
+@patch('os.path.isdir', lambda x: False)
+@patch('tarfile.TarFile.extractall')
+def test_download_and_extract_tar(extractall, s3_download):
+    t = tarfile.open(name='test.tar.gz', mode='w:gz')
+    t.close()
+    uri = t.name
+    _files.download_and_extract(uri, _env.code_dir)
+
+    s3_download.assert_not_called()
+    extractall.assert_called()
```

Comment:
use assert_called_with with the expect arguments.

Reply:
sounds good